### PR TITLE
Button and icon with corresponding color logic

### PR DESCRIPTION
**Pull request by britt6612**
#### What does this PR do?
Adds logic to Button to take color of Icon into account
#### Where should the reviewer start?
Button/button.js
#### What testing has been done on this PR?
manually
#### How should this be manually tested?
storybook
#### What are the relevant issues?
#2820 
#### Do the grommet docs need to be updated?
auto
#### Should this PR be mentioned in the release notes?
no
#### Is this change backwards compatible or is it a breaking change?
backwards compatible

### CODE REVIEW

**Comment by britt6612** [comment]
@ShimiSun @ericsoderberghp rockin:

**Comment by ShimiSun** [comment]
Screen Shot 2019-07-26 at 3 29 08 PM

This is a confirmation this branch plays nice with the master's snapshots ✅

**Comment by ShimiSun** [comment]
More than 5 snapshots had changed, which may result in a backward compatibility issue.

@britt6612 can you create a new story for the Button component that will demonstrate all the different ways the button can apply the colors, e.g. primary, simple, w/o icon, w/o theme or whatever else you find relevant for the new logic applied. (similar story reference is the https://storybook.grommet.io/?path=/story/button--icon-plain), thank you for working on this!
Creating the story will help us see the different use cases and understand the potential risks of changing this fundamental component.

File: src/js/components/Button/stories/IconColor.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  </Grommet>
+);
+
+const IconLabelColor = () => (
+  <Grommet theme={grommet}>
+    <Box align="center" pad="large">
+      <Box align="center" pad="large" gap="small">
+        <Button icon={<Add />} label="Add" onClick={() => {}} primary />
+        <Button icon={<Add />} label="Add" onClick={() => {}} />
+        <Button icon={<Add />} label="Add" gap="xlarge" onClick={() => {}} />
```
Comment (ShimiSun) [inline src/js/components/Button/stories/IconColor.js]: I think lines 34 & 35 can be replaced with showing different cases of only an icon and only a label and see how it behaves. so we can compare the use cases.

**Comment by britt6612** [comment]
@ShimiSun going to do this in another branch easier to have updated master with these changes

**Pull request closed without merging**